### PR TITLE
Add userVisibleData, userVisibleDataFormat, and userNonVisibleData support to `.authenticate`

### LINKED ISSUE #69: Support of `userVisibleData` and `userVisibleDataFormat` to `.authenticate`?
Hi! The [BankID api for /auth](https://www.bankid.com/utvecklare/guider/teknisk-integrationsguide/graenssnittsbeskrivning/auth) supports `userVisibleData` and `userVisibleDataFormat` to show the user some markdown when performing authentication.

As far as I can tell these parameters are not supported in this library. Do you have any plans to add support for these parameters?

### PR DESCRIPTION
**Pull request by alepek**
These variables were already technically supported but were hard to use due to the types not containing them. I've also updated the tests to reflect this change, and verified the functionality with my own test BankID.

closes #69

### CODE REVIEW

File: test/manual/src/manualTests.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { BankIdClient } from "bankid";
+import { BankIdClient } from "../../../";
```
Comment (alepek) [inline test/manual/src/manualTests.ts:1]: Had to make this change for the test to run. Possibly my local env is misconfigured, or the package.json root config is?

File: src/bankid.ts [hunk 2]
```diff
@@ -36,7 +38,6 @@ interface AuthOptionalRequirements {
 
 export interface SignRequest extends AuthRequest {
   userVisibleData: string;
-  userVisibleDataFormat?: "simpleMarkdownV1";
```
Comment (pmmmwh) [inline src/bankid.ts:39]: Since we're changing this - we might as well move `userNonVisibleData` to `AuthRequest` as well.

File: test/manual/src/manualTests.ts (hunk 1 above)
Comment (pmmmwh) [reply test/manual/src/manualTests.ts:1]: These tests are suppose to run with a "linked" local BankID build, so let's revert.

```suggestion
import { BankIdClient } from "bankid";
```

File: test/manual/yarn.lock [hunk 3]
```diff
@@ -12,18 +12,19 @@ asynckit@^0.4.0:
...
+  integrity sha512-Ii012v05KEVuUoFWmMW/UQv9aRIc3ZwkWDcM+h5Il8izZCtRVpDUfwpoFf7eOtajT3QiGR4yDUx7lPqHJULgbg==
   dependencies:
-    follow-redirects "^1.14.9"
+    follow-redirects "^1.15.4"
     form-data "^4.0.0"
+    proxy-from-env "^1.1.0"
 
 "bankid@file:../..":
-  version "3.1.2"
+  version "3.1.3"
```
Comment (pmmmwh) [inline test/manual/yarn.lock:25]: This line we should keep.

File: test/manual/yarn.lock [hunk 4]
```diff
@@ -12,18 +12,19 @@ asynckit@^0.4.0:
   resolved "https://registry.yarnpkg.com/asynckit/-/asynckit-0.4.0.tgz#c79ed97f7f34cb8f2ba1bc9790bcc366474b4b79"
   integrity sha512-Oei9OH4tRh0YqU3GxhX79dM/mwVgvbZJaSNaRk+bshkj0S5cfHcgYakreBjrHwatXKbz+IoIdYLxrKim2MjW0Q==
 
-axios@^0.27.2:
-  version "0.27.2"
-  resolved "https://registry.yarnpkg.com/axios/-/axios-0.27.2.tgz#207658cc8621606e586c85db4b41a750e756d972"
-  integrity sha512-t+yRIyySRTp/wua5xEr+z1q60QmLq8ABsS5O9Me1AsE5dfKqgnCFzwiCZZ/cGNd1lq4/7akDWMxdhVlucjmnOQ==
+axios@^1.6.4:
```
Comment (pmmmwh) [inline test/manual/yarn.lock:15]: We've been putting off upgrading Axios due to potential breaking changes, so I would suggest we revert the changes here.

**Review by pmmmwh — CHANGES_REQUESTED** [review]

**Comment by pmmmwh** [comment]
Thanks for the PR 😄

**Comment by alepek** [comment]
Thanks for the feedback! Will address it later today.

File: test/manual/yarn.lock (hunk 4 above)
Comment (alepek) [reply test/manual/yarn.lock]: I reverted all changes to the lockfile. Worth nothing is that the will reappear upon running `yarn` in the test project.

File: src/bankid.ts [hunk 5]
```diff
@@ -239,6 +240,26 @@ export class BankIdClient {
...
     }
+    if (
+      parameters.userVisibleDataFormat &&
+      parameters.userVisibleDataFormat !== "simpleMarkdownV1"
+    ) {
+      throw new Error("userVisibleDataFormat can only be simpleMarkdownV1.");
+    }
+
+    console.log(parameters);
+
```
Comment (pmmmwh) [inline src/bankid.ts:251]:
```suggestion
```

File: src/bankid.ts [hunk 6]
```diff
@@ -239,6 +240,26 @@ export class BankIdClient {
...
+      ...parameters,
+      userVisibleData: parameters.userVisibleData
+        ? Buffer.from(parameters.userVisibleData).toString("base64")
+        : undefined,
+      userNonVisibleData: parameters.userNonVisibleData
+        ? Buffer.from(parameters.userNonVisibleData).toString("base64")
+        : undefined,
+    };
+
+    console.log(parameters);
```
Comment (pmmmwh) [inline src/bankid.ts:262]:
```suggestion
```

File: test/manual/src/manualTests.ts [hunk 7]
```diff
@@ -17,37 +17,33 @@ async function main() {
     production: false,
   });
 
-  try {
```
Comment (pmmmwh) [inline test/manual/src/manualTests.ts:20]: I think it's nice to be able to run all of the tests here - so if auth fails we can still see if sign works - maybe what was missing is some way to track "errored" status?

Comment (alepek) [reply test/manual/src/manualTests.ts:20]: Fair enoguh! I can update it to produce a summary of some sort at the end.

**Comment by alepek** [comment]
Thanks for the feedback and pardon the slightly sloppy remnants of my debugging. Will adjust according to feedback and get back to you.

**Comment by alepek** [comment]
I've changed the source code according to feedback and updated the test suite to output success/failure messages for each separate test and a summary at the end. 

Example successful run
Screenshot 2024-01-23 at 19 05 55

Example failed run (manual cancel of first two tests)
Screenshot 2024-01-23 at 19 06 54

File: test/manual/src/manualTests.ts [hunk 8]
```diff
@@ -74,29 +80,40 @@ async function main() {
...
-  .then(() => console.log("✅ All tests completed successfully"))
-  .catch(err => console.log(err));
+main().then(errors => {
+  const failedTests = errors.length;
+  if (failedTests === 0) {
+    console.log("✅ All tests completed successfully");
+  } else {
+    console.error(`❌ ${failedTests} test(s) failed`);
+  }
+});
```
Comment (pmmmwh) [inline test/manual/src/manualTests.ts:116]: Food for thought - should we keep a `.catch` here just in case? (Skimmed through the test code and I believe the only thing that can throw is client instantiation or `delay` 🤔 )

**Comment by pmmmwh** [comment]
Looks great! I have only one more thought 🙇🏻

File: test/manual/src/manualTests.ts (hunk 8 above)
Comment (pmmmwh) [inline test/manual/src/manualTests.ts:116]:
```suggestion
main()
	.then(errors => {
	  const failedTests = errors.length;
	  if (failedTests === 0) {
	    console.log("✅ All tests completed successfully");
	  } else {
	    console.error(`❌ ${failedTests} test(s) failed`);
	  }
	})
	.catch(console.error);
```

Comment (alepek) [reply test/manual/src/manualTests.ts]: Yes, that is a fair point. Committed your suggestion!

**Review by pmmmwh — APPROVED** [review]
Thank you! I'll try to cut a release this week.
